pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -1620,7 +1620,12 @@ Future<Result, MessageId> ClientConnection::newGetLastMessageId(uint64_t consume
 
     pendingGetLastMessageIdRequests_.insert(std::make_pair(requestId, promise));
     lock.unlock();
-    sendRequestWithId(Commands::newGetLastMessageId(consumerId, requestId), requestId);
+    sendRequestWithId(Commands::newGetLastMessageId(consumerId, requestId), requestId)
```

Comment:
I think we should keep this change separated from the below refactoring of the Future class.

Reply:
I agree. I'll remove the unnecessary refactoring of this class but still add some changes.